### PR TITLE
changes weekly allocation hour estimate calculation

### PR DESCRIPTION
**Pull request by jduss4**
## Description

Previously, we changed `total_allocation_hours` (an estimate of how many hours a person with weekly allocation % spent working on the project) to use `target_hours` in its calculation. This worked great for individuals with a .8 billable expectation, but not so well for folks like supervisors with a .4 billable expectation, as it then halved their potentially contributed hours.

Here is a new proposal for the `total_allocation_hours` estimate that returns to the assumption that there is a set number of hours that people would be likely to bill during the week (current values are 32 hours / .8 billable expectation, set in `base.py`), and that a person's allocation % is a percentage of THAT time.  Therefore, we use the default billable expectation to compare allocation %s against.

This adds a test for a lower billable expectation (.4) with out of office time, and checks the allocation hour estimate in several other tests.

## Additional information

We have gone back and forth a little on how to consider utilization for supervisors.  Is a supervisor with a .4 expectation (16 hours a week) who bills 50% (16 hours) to a weekly project considered 50% utilized or 100% utilized?  The current utilization behavior is that a supervisor billing 50% would be 50% utilized (out of their entire week), but if this is incorrect, then we need to revisit how utilization is being calculated.

### CODE REVIEW

**Comment by jduss4** [comment]
We are hoping to get @kfoley-18F 's sign-off or make adjustments to the logic based on her review before we can make it a real PR!

File: tock/hours/models.py
Comment (kfoley-18F) [inline tock/hours/models.py:333]: Yes, this explanation of WB allocation instead of hourly expectation aligns with my understanding too.

**Review by kfoley-18F — APPROVED** [review]
Yes!

**Comment by jduss4** [comment]
Thanks for your review, @kfoley-18F !

Would you mind double checking our understanding of this question, as well?

> Is a supervisor with a .4 expectation (16 hours a week) who bills 50% (16 hours) to a weekly project considered 50% utilized or 100% utilized? The current utilization behavior is that a supervisor billing 50% would be 50% utilized (out of their entire week), but if this is incorrect, then we need to revisit how utilization is being calculated.

**Comment by jduss4** [comment]
@cantsin ready for code review at your convenience! Thank you!

**Comment by kingcomma** [comment]
@kfoley-18F If it's helpful for answering this question, we put together some [example scenarios in this doc](https://docs.google.com/document/d/1lMSumCAfLzpTS2uKmDJPfe5rM6swEF9LWIB89leTTMs/edit#heading=h.7uhyk4kppumv). I've highlighted the places we have questions and added comments to explain our uncertainty.

> Thanks for your review, @kfoley-18F !
> 
> Would you mind double checking our understanding of this question, as well?
> 
> > Is a supervisor with a .4 expectation (16 hours a week) who bills 50% (16 hours) to a weekly project considered 50% utilized or 100% utilized? The current utilization behavior is that a supervisor billing 50% would be 50% utilized (out of their entire week), but if this is incorrect, then we need to revisit how utilization is being calculated.

**Review by cantsin — APPROVED** [review]
Love the test! Code LGTM; will wait for Kelsey before merging.

**Comment by cantsin** [comment]
Yay! Builds are failing for an entirely unrelated reason (we need to bump up the `CACHE_VERSION` in CircleCI) so let me quickly fix that and then, once builds pass again, we'll merge!

**Pull request merged**
